orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/orca/q/redis/RedisDeadLetterRedriveCommand.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.q.redis
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.pipeline.model.Orchestration
+import com.netflix.spinnaker.orca.pipeline.model.Pipeline
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import com.netflix.spinnaker.orca.q.*
+import com.netflix.spinnaker.orca.q.admin.DeadLetterRedriveCommand
+import org.slf4j.LoggerFactory
+import redis.clients.jedis.Jedis
+import redis.clients.util.Pool
+
+/**
+ * Offers the ability to redrive a range of dead-lettered messages back onto the queue.
+ *
+ * We currently do not support re-driving RunTask messages, for fear of non-idempotent operations;
+ * although this should be trimmed down over time.
+ */
+class RedisDeadLetterRedriveCommand(
+  deadLetterQueueName: String,
+  private val queue: Queue,
+  private val pool: Pool<Jedis>,
+  private val executionRepository: ExecutionRepository
+) : DeadLetterRedriveCommand {
+
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  private val dlqKey = "$deadLetterQueueName.messages"
+
+  private val mapper = ObjectMapper().apply {
+    registerModule(KotlinModule())
+  }
+
+  override fun redrive(after: Long?, before: Long?) {
+    pool.resource.use { redis ->
+      val redrivenMessages = getMessages(redis, after, before)
+
+      log.warn("Redriving ${redrivenMessages.size} dead lettered messages between ${after?:"-inf"} and ${before?:"+inf"}")
```

Comment:
I think at least one of [before,after] should be a required argument instead of defaulting to *. Redriving overly old messages could have unintended negative consequences to production services managed by Spinnaker. I think that justifies forcing the operator to consider the time range of DLQ'd messages to redrive.